lib/blocklist.c

```diff
@@ -102,8 +104,33 @@ void allowlist_prefix(char *ip, int prefix_len)
 	_add_constraint(addr, prefix_len, ADDR_ALLOWED);
 }
 
+static int is_ip_ipv6(char *ip) {
+	// don't modify the input string
+	int length = strlen(ip);
+	char *new_str = (char*)malloc(length);
```

Comment:
This should use `xmalloc`, not `malloc`. This also makes the following check unnecessary.